test/test_autograd.py

```diff
@@ -2174,6 +2174,19 @@ def func(x):
         gradgradcheck(func, [torch.rand(m).add_(1).requires_grad_()])
         gradgradcheck(func, [torch.rand(m).add_(10).requires_grad_()])
 
+    @skipIfRocm
+    def test_chain_matmul(self):
+        def gen_matrices(p):
+            matrices = []
+            for (pi, pi_1) in zip(p[:-1], p[1:]):
+                matrices.append(torch.randn(pi, pi_1).requires_grad_())
+            return matrices
+
+        gradcheck(torch.chain_matmul, gen_matrices([5, 10, 15, 5]))
```

Comment:
add a test for 3 matrices since you specialized the impl.

Reply:
Same as above, this is covered. However, I will add a case for N > 3.